openedx/features/qverse_features/registration/validators.py

```diff
@@ -1,31 +1,65 @@
 """
 Validation utils for qverse registration application.
 """
+from csv import reader
+import io
+
 from django.core.exceptions import ValidationError
 
 
-def validate_file_extension(file):
+def validate_admission_file(file):
     """
-    Validates that file is a valid csv file by checking its extension.
+    Validates that the file is a valid csv file by checking its extension.
+
+    Also checks that the header row has all the required fields.
+
+    Arguments:
+        file - (models.FieldFile) The File uploaded by the user
+
+    Returns:
+        None
     """
     if not file.name.endswith('.csv'):
         raise ValidationError('', code='invalid')
 
+    FIELD_NAMES = [
+                'regno', 'firstname', 'surname', 'othername', 'levelid',
+                'programmeid', 'departmentid', 'mobile', 'email', 'mobile'
+                ]
+    decoded_file = file.read().decode('utf-8')
+    io_string = io.StringIO(decoded_file)
+    header_row = reader(io_string, delimiter=',').next()
+    header_row = [heading.lower().strip() for heading in header_row]
+    if not all(field_name in header_row for field_name in FIELD_NAMES):
+        raise ValidationError('', code='invalid')
+
 
 def validate_current_level(value):
     """
     Validates the provided current level.
+
+    Arguments:
+        value - (int) Current educational year at school
+
+    Returns:
+        None
     """
     # to avoid circular import
     from openedx.features.qverse_features.registration.models import MAX_LEVEL_CHOICES
 
-    if not 1 <= value <= MAX_LEVEL_CHOICES:
+    if not 1 <= value < MAX_LEVEL_CHOICES:
```

Comment:
```suggestion
    if not value in range(1, MAX_LEVEL_CHOICES):
```

Improves user readability and understanding. It took me 2 to 3 mins to understand your condition.

Reply:
done.